portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/ABTestACSide.testcase

```diff
@@ -0,0 +1,363 @@
+	@priority = 5
+	test CanPublishControlFromTerminatedTest {
+		task ("Edit the variant created") {
+			ABTest.editVariant(variantName = "Variant name");
+		}
+
+		task ("Add a heading to the page to diferentiate from the control") {
+			PageEditor.addFragment(
+				collectionName = "Basic Components",
+				fragmentName = "Heading");
+		}
+
+		task ("Click to save the variant") {
+			ABTest.saveVariant();
+		}
+
+		task ("Click to open the AB test panel") {
+			ABTest.openABTestPanel();
+
+			task ("Click to review and run the test") {
+				ABTest.runTest();
+			}
+		}
+
+		task ("Click to terminate the test") {
+			Button.click(button = "Terminate Test");
+
+			ABTest.assertTerminateTest();
+		}
+
+		task ("Click to publish the control") {
+			ABTest.publishVariant(variantName = "Control");
+		}
+
+		task ("Assert the alert is shown") {
+			Alert.viewSuccessMessageText(successMessage = "Success:Control was published successfully.");
+		}
+
+		task ("Assert the control is published") {
+			AssertElementNotPresent(
+				locator1 = "Experience#FRAGMENT_HEADING",
+				value1 = "Heading Example");
+		}
+	}
+
+	@description = "Story: LRAC-14070 | Automation ID: LRAC-14573 | Test Summary: Can publish variant from a terminated test and delete test in AC"
+	@priority = 5
+	test CanPublishVariantFromTerminatedTestAndDeleteTest {
+		task ("Edit the variant created") {
+			ABTest.editVariant(variantName = "Variant name");
+		}
+
+		task ("Add a heading to the page to diferentiate from the control") {
+			PageEditor.addFragment(
+				collectionName = "Basic Components",
+				fragmentName = "Heading");
+		}
+
+		task ("Click to save the variant") {
+			ABTest.saveVariant();
+		}
+
+		task ("Click to open the AB test panel") {
+			ABTest.openABTestPanel();
+
+			task ("Click to review and run the test") {
+				ABTest.runTest();
+			}
+		}
+
+		task ("Click to terminate the test") {
+			Button.click(button = "Terminate Test");
+
+			ABTest.assertTerminateTest();
+		}
+
+		task ("Click to publish the variant") {
+			ABTest.publishVariant(variantName = "Variant name");
+		}
+
+		task ("Assert the alert is shown") {
+			Alert.viewSuccessMessageText(successMessage = "Success:Variant name was published successfully.");
+		}
+
+		task ("Assert the variant published") {
+			ContentPages.viewFragmentText(
+				fragmentName = "heading",
+				id = "element-text",
+				text = "Heading Example");
+		}
+
+		task ("Go to AC > Tests") {
+			ACUtils.launchAC();
+
+			ACProperties.switchProperty(propertyName = ${assignedPropertyName});
+
+			ACNavigation.goToTests();
+		}
+
+		task ("Open the AB test") {
+			ACNavigation.openItem(itemName = ${nameABTest});
+		}
+
+		task ("Delete the test in AC") {
+			ABTest.deleteABTestACSide();
+		}
+
+		task ("Assert the test is no longer present in the list") {
+			ACUtils.assertItemsNameNotPresent(itemList = ${nameABTest});
+		}
+	}
```

Comment:
Hey @manuferreira, When I created the files for the AB test, I thought this way:

- **ABTestACSide.testcase** - Let's put test cases here that have the purpose of checking something in the AB test part on the AC side. So the test cases that have AC as the main

- **ABTestDXPSide.testcase** - We will place test cases here that have the purpose of checking something in the AB test part on the DXP side. So the test cases that have DXP as the main

I thought about this division to prevent us from having a very large file with all the test cases and making it difficult to look at the code. I wanted to know if you agree with this division? If yes, then I think these two test cases should be moved to `ABTestDXPSide.testcase` since their main purpose is the publish that occurs DXP

Reply:
I saw you put together two automations in a single case. I think you can relate the two tickets in Jira and close one of them. This way, you can update one of them with the correct steps that will cover both cases and put the new test name on the ticket. I think you are still using the old name on the ticket

**Test Case:** CanPublishVariantFromTerminatedTestAndDeleteTest

Reply:
https://github.com/liferay-ac/liferay-portal/pull/729/commits/7d002271ba825581b216062faf7584d9a274a33f